### PR TITLE
[T4A3][T16-A1]Lai Zhiyuan

### PR DESCRIPTION
**Pull request by zyjames**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/data/person/Contact.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represent a Person's contact in the address book.
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:4]: good! but not necessary to say address book; you can use this class in any application that deals with contact information :-)

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ */
+public class Contact {
+
+	public final String value;
+    private boolean isPrivate;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:9]: indentation is off here!

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represent a Person's contact in the address book.
+ */
+public class Contact {
+
+	public final String value;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:8]: try to make it private. in general, start off with private members, then use protected or public only if absolutely necessary.

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -6,27 +6,22 @@
...
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        String trimmedAddress = address.trim();
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        super(address.trim(), isPrivate);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java:21]: in this case you are assigning the values before validation; while this works here, think of ways to mitigate this problem.

**Comment by okkhoy** [comment]
@zyjames 
Some comments added. Please ack & close the PR after reading comments.

File: src/seedu/addressbook/data/person/Contact.java (hunk 3 above)
Comment (zyjames) [reply src/seedu/addressbook/data/person/Contact.java:8]: Thanks for pointing out my mistake!(:

File: src/seedu/addressbook/data/person/Contact.java (hunk 2 above)
Comment (zyjames) [reply src/seedu/addressbook/data/person/Contact.java:9]: I will take note in future!

File: src/seedu/addressbook/data/person/Address.java (hunk 4 above)
Comment (zyjames) [reply src/seedu/addressbook/data/person/Address.java:21]: I think a good way is to pass in the validation regex and validate before assignment

**Pull request closed without merging**